fabtests/include/shared.h

```diff
@@ -520,14 +525,43 @@ static inline uint64_t ft_gettime_ms(void)
 	return ft_gettime_ns() / 1000000;
 }
 
-static inline void ft_start(void)
+static inline void ft_timer_start(union ft_timer *timer)
+{
+	timer->start = ft_gettime_ns();
+}
+static inline void ft_timer_pause(union ft_timer *timer)
+{
+	timer->elapsed = ft_gettime_ns() - timer->start;
+}
+static inline void ft_timer_resume(union ft_timer *timer)
+{
+	timer->start = ft_gettime_ns() - timer->elapsed;
+}
+static inline void ft_timer_stop(union ft_timer *timer)
+{
+	ft_timer_pause(timer);
+}
+static inline bool ft_timer_is_elapsed(union ft_timer timer, uint64_t timeout, enum precision p)
+{
+	// assuming timer is running
```

Comment:
typo, also couldn't hurt to add some asserts in the timer instead of having to comment this

Reply:
Ah I missed the union! A fun optimization! But I think I would prefer the struct, especially if that adds the option of an assert to catch misuses of it. I wouldn't want performance numbers to get reported incorrectly if someone forgot to stop/resume in the correct order or something.